test/provisioner/provision_ibmcloud_kustomize.go

```diff
@@ -170,7 +171,11 @@ func (lio *IBMCloudInstallOverlay) Edit(ctx context.Context, cfg *envconf.Config
 			}
 		}
 	}
-
+	if os.Getenv("REGISTRY_CREDENTIAL_ENCODED") != "" {
+		if err = lio.overlay.SetKustomizeSecretGeneratorFile("auth-json-secret", "auth.json"); err != nil {
```

Comment:
I think this code might need updating to either clean up this kustomize, or check if it already exists?. I tried to run the test cases locally and ended up with:
```
F0710 08:20:49.598761   12477 env.go:369] Setup failure: configmap auth-json-secret illegally repeats the key `auth.json`
```

and looking at `kustomization.yaml` I could see:
```
secretGenerator:
- files:
  - auth.json
  - auth.json
  name: auth-json-secret
  namespace: confidential-containers-system
```

Reply:
I think that the issue is actually in `test/provisioner/kustomize.go:SetKustomizeSecretGeneratorFile`, so that needs a closer look.

Reply:
I'm also getting:
```
=== RUN   TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test
    common_suite_test.go:135: secrets "auth-json-secret" already exists
```
on repeated tests, so maybe we need to consider cleaning this up in the teardown too?

Reply:
I'm hitting:
```
=== RUN   TestCreatePeerPodWithAuthenticatedImagewithValidCredentials
=== RUN   TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test
    common_suite_test.go:161: timed out waiting for the condition
--- FAIL: TestCreatePeerPodWithAuthenticatedImagewithValidCredentials (600.20s)
    --- FAIL: TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test (600.20s)
=== RUN   TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials
=== RUN   TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials/InvalidAuthImagePeerPod_test
    common_suite_test.go:135: secrets "auth-json-secret" already exists
--- FAIL: TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials (0.06s)
    --- FAIL: TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials/InvalidAuthImagePeerPod_test (0.06s)
FAIL
FAIL	github.com/confidential-containers/cloud-api-adaptor/test/e2e	2308.938s
FAIL
```
now - the the authenticated image not being cleaned up is still an issue?

Reply:
In this case,
It seems in `TestCreatePeerPodWithAuthenticatedImagewithValidCredentials` , the pod didnt reached `Running/Completed` state, so test case got terminated in Setup method without reaching Assess & Teardown
So while running next test case `TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials`, which also creates auth-json-secret faced a conflict.

Reply:
Hi @stevenhorsman ,
- Updated code logic to solve the above addressed issue
- Also I tried to recreate your case by passing invalid credentials, to make `TestCreatePeerPodWithAuthenticatedImagewithValidCredentials ` FAIL and now auth-json-secret gets deleted 

```
=== RUN   TestCreatePeerPodWithAuthenticatedImagewithValidCredentials
=== RUN   TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test
=== RUN   TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test/Peer_pod_with_Authenticated_Image_with_Valid_Credentials_has_been_created
    common_suite_test.go:226: Invalid Credentials
time="2023-07-13T00:48:25+05:30" level=info msg="Deleting Secret... auth-json-secret"
time="2023-07-13T00:48:26+05:30" level=info msg="Deleting pod... authenticated-image-valid-951-pod"
--- FAIL: TestCreatePeerPodWithAuthenticatedImagewithValidCredentials (23.63s)
    --- FAIL: TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test (23.63s)
        --- FAIL: TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test/Peer_pod_with_Authenticated_Image_with_Valid_Credentials_has_been_created (17.09s)
=== RUN   TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials
=== RUN   TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials/InvalidAuthImagePeerPod_test
=== RUN   TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials/InvalidAuthImagePeerPod_test/Peer_pod_with_Authenticated_Image_with_Invalid_Credentials_has_been_created
    common_suite_test.go:229: PodVM has successfully reached ImagePullBackOff state with authenticated Image - quay.io/sudharshanibm3/authenticated-registry:latest
time="2023-07-13T00:48:49+05:30" level=info msg="Deleting Secret... auth-json-secret"
time="2023-07-13T00:48:50+05:30" level=info msg="Deleting pod... authenticated-image-invalid-295-pod"
--- PASS: TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials (24.05s)
    --- PASS: TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials/InvalidAuthImagePeerPod_test (24.05s)
        --- PASS: TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials/InvalidAuthImagePeerPod_test/Peer_pod_with_Authenticated_Image_with_Invalid_Credentials_has_been_created (17.50s)
```

Could you please try now?

Reply:
Thanks, I'm seeing the same thing now:
```
=== RUN   TestCreatePeerPodWithAuthenticatedImagewithValidCredentials
=== RUN   TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test
=== RUN   TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test/Peer_pod_with_Authenticated_Image_with_Valid_Credentials_has_been_created
    common_suite_test.go:226: Invalid Credentials
time="2023-07-13T02:24:51-07:00" level=info msg="Deleting Secret... auth-json-secret"
time="2023-07-13T02:24:51-07:00" level=info msg="Deleting pod... authenticated-image-valid-81-pod"
--- FAIL: TestCreatePeerPodWithAuthenticatedImagewithValidCredentials (24.19s)
    --- FAIL: TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test (24.19s)
        --- FAIL: TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test/Peer_pod_with_Authenticated_Image_with_Valid_Credentials_has_been_created (18.94s)
=== RUN   TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials
=== RUN   TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials/InvalidAuthImagePeerPod_test
=== RUN   TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials/InvalidAuthImagePeerPod_test/Peer_pod_with_Authenticated_Image_with_Invalid_Credentials_has_been_created
    common_suite_test.go:229: PodVM has successfully reached ImagePullBackOff state with authenticated Image - quay.io/stevenhorsman/busybox
time="2023-07-13T02:25:13-07:00" level=info msg="Deleting Secret... auth-json-secret"
time="2023-07-13T02:25:13-07:00" level=info msg="Deleting pod... authenticated-image-invalid-887-pod"
--- PASS: TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials (22.09s)
    --- PASS: TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials/InvalidAuthImagePeerPod_test (22.08s)
        --- PASS: TestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials/InvalidAuthImagePeerPod_test/Peer_pod_with_Authenticated_Image_with_Invalid_Credentials_has_been_created (16.86s)
```

I still can't get the valid credentials use case to work and I've tried two different repos and two different sets of credentials, so can you confirm that I'm doing things correctly? I'm setting `export REGISTRY_CREDENTIAL_ENCODED="base64 (username:password)"` which I got from copy pasting the docker config from quay.io.

Reply:
Yeah seems like a correct way
Can you try clearing auth-json-secret manually from `confidential-containers-system` namespace and try again